src/main.rs

```diff
@@ -95,8 +96,16 @@ fn main() {
    );
 
    let json = serde_json::to_string(&reading).unwrap();
+   let url = "http://localhost:4000/api/readings/add";
 
-   println!("{}", json);
+   println!("Sending POST request to {} as JSON: {}", url, json);
+
+   let client = Client::new();
+   let res = client.post(url)
+    .body(json)
+    .send();
+
```

Comment:
I'm curious to learn with this application. What are some ways to deal with error handling in this case that comes through with `res`?

Reply:
Thats a really good question. I looked up the function signature for `send()` and it says [here](https://docs.rs/reqwest/latest/reqwest/blocking/struct.RequestBuilder.html#method.send) that it returns a `Result` enum. I decided to just print out whatever the return from `send()` was for now so that we can have a visual on what the returns look like. 

Here's an example where I turned off the Ambi server 

```
Response: Err(. # this part
    reqwest::Error {
        kind: Request,
        url: Url {
            scheme: "http",
            cannot_be_a_base: false,
            username: "",
            password: None,
            host: Some(
                Domain(
                    "localhost",
                ),
            ),
            port: Some(
                4000,
            ),
            path: "/api/readings/add",
            query: None,
            fragment: None,
        },
        source: hyper::Error(
            Connect,
            ConnectError(
                "tcp connect error",
                Os {
                    code: 61,
                    kind: ConnectionRefused,
                    message: "Connection refused",
                },
            ),
        ),
    },
)
```

That `Err()` enum value that surrounds the response data is the `E` option of the `Result` whereas when it works

```
Ok( # this part
    Response {
        url: Url {
            scheme: "http",
            cannot_be_a_base: false,
            username: "",
            password: None,
            host: Some(
                Domain(
                    "localhost",
                ),
            ),
            port: Some(
                4000,
            ),
            path: "/api/readings/add",
            query: None,
            fragment: None,
        },
        status: 200,
        headers: {
            "cache-control": "max-age=0, private, must-revalidate",
            "content-length": "60",
            "content-type": "application/json; charset=utf-8",
            "date": "Tue, 25 Jan 2022 00:29:33 GMT",
            "server": "Cowboy",
            "x-request-id": "Fs1cB8syTfumks8AAABh",
        },
    },
)
```

The response data is wrapped in `Ok()`. 

So you could do something like they recommend in the [Result](https://doc.rust-lang.org/std/result/) docs and use a `match`

something like 

```Rust
let res = client.post(url)
    .body(json)
    .send();

match res {
  Ok(data) => println!("Successful Response: {:#?}", data),
  Err(error_data) => println!("Error Response: {:#?}", error_data)
}

```

https://doc.rust-lang.org/std/result/

Reply:
oh nice. that's great to know. Yeah, I was just curious of how it was incorporated. I'll keep that in mind. Thank you Caleb.